.github/workflows/c-cpp.yml

```diff
@@ -1,17 +1,93 @@
-name: C/C++ CI
+name: nostril
 
+# Controls when the action will run.
 on:
-  push:
-    branches: [ "master", "**" ]
   pull_request:
-    branches: [ "master" ]
+    branches:       #[ master ]
+      - '*'         # matches every branch that doesn't contain a '/'
+      - '*/*'       # matches every branch containing a single '/'
+      - '**'        # matches every branch
+      - 'master'    # includes master
+      - 'main'      # includes main
+  push:
+    paths:
+      - '.github/workflows/c-cpp.yml'
+    schedule:
+      - cron: "1 * * * *"
+    branches:       #[ master ]
+      - '*'         # matches every branch that doesn't contain a '/'
+      - '*/*'       # matches every branch containing a single '/'
+      - '**'        # matches every branch
+      - 'master'    # includes master
+      - 'main'      # includes main
 
-jobs:
-  build:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
 
-    runs-on: ubuntu-latest
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  github-workspace:
+    name: github-workspace
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+      - name: Print github workspace
+        run: |
+          printenv
 
+  ubuntu-build_and_test:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v3
-    - name: make 
-      run: make 
+      - uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v2
+        #python venv stub for other scripting/tests
+      - uses: actions/setup-python@v2
```

Comment:
I created a python3 venv so that additional testing can be scripted with python3.